servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -240,11 +249,56 @@ public StreamingHttpClient apply(final UrlKey urlKey) {
                 builder.sslConfig(DEFAULT_CLIENT_SSL_CONFIG);
             }
 
+            MultiAddressStrategyWrapper wrapper = new MultiAddressStrategyWrapper();
+            builder.appendClientFilter(wrapper);
+
             if (singleAddressInitializer != null) {
                 singleAddressInitializer.initialize(urlKey.scheme, urlKey.hostAndPort, builder);
             }
 
-            return builder.buildStreaming();
+            StreamingHttpClient singleClient = builder.buildStreaming();
+
+            wrapper.clientStrategy = singleClient.executionContext().executionStrategy();
+            return singleClient;
+        }
+    }
+
+    private static final class MultiAddressStrategyWrapper implements StreamingHttpClientFilterFactory {
+
+        HttpExecutionStrategy clientStrategy = defaultStrategy();
+        @Override
+        public StreamingHttpClientFilter create(final FilterableStreamingHttpClient client) {
+            return new StreamingHttpClientFilter(client) {
+                @Override
+                protected Single<StreamingHttpResponse> request(
+                        final StreamingHttpRequester delegate, final StreamingHttpRequest request) {
+                    applyMultiAddressStrategy(request.context(), clientStrategy);
```

Comment:
1. I believe we should wrap it with `defer` to make sure we check the strategy on each re-subscribe.
2. You can take `clientStrategy` from `client.executionContext().executionStrategy()`, no need to have a class-level variable.